symfony/src/Controller/DeceasedReportsController.php

```diff
@@ -0,0 +1,124 @@
+        $report = (new DeceasedReport())->loadFromFhirObservation($fhirData);
+        if ($report->getSubmittedBy() === $this->getUser()->getEmail() && $report->getReportStatus() === 'preliminary') {
+            $this->addFlash('notice', 'You submitted this report. Another user must review it.');
+        }
+        $form = $this->createForm(DeceasedReportReviewType::class, $report, ['reviewer_email' => $this->getUser()->getEmail()]);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $data = $form->getData();
+            try {
+                $fhirData = $deceasedReportsService->buildDeceasedReportReviewFhir($data, $this->getUser());
+                $deceasedReportsService->updateDeceasedReport($participantId, $reportId, $fhirData);
+                $report->setReportStatus('preliminary');
+                $this->addFlash('success', 'Report updated!');
+                return $this->redirectToRoute('deceased_reports_index');
+            } catch (\Exception $e) {
+                error_log($e->getMessage());
+                $report->setReportStatus('preliminary');
+                $this->addFlash('error', sprintf('Invalid API response [%s]', $e->getCode()));
+            }
+        }
+        return $this->render('deceasedreports/review.html.twig', [
+            'report' => $report,
+            'participant' => $participant,
+            'form' => $form->createView()
+        ]);
+    }
+
+    /**
+     * @Route("/{participantId}/new", name="deceased_report_new", requirements={"participantId"="P\d+"})
+     */
+    public function deceasedReportNew(Request $request, ParticipantSummaryService $participantSummaryService, DeceasedReportsService $deceasedReportsService, $participantId) {
+        $participant = $participantSummaryService->getParticipantById($participantId);
+        if (!$participant) {
+            throw $this->createNotFoundException('Participant not found.');
+        }
+        $reports = $deceasedReportsService->getDeceasedReportsByParticipant($participantId);
+        $report = new DeceasedReport();
+        foreach ($reports as $record) {
+            if ($record->status === 'cancelled') {
+                continue;
+            }
+            $report = (new DeceasedReport())->loadFromFhirObservation($record);
+        }
+        $form = $this->createForm(DeceasedReportType::class, $report, ['disabled' => (bool) $report->getId()]);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $report = $form->getData();
+            try {
+                $fhirData = $deceasedReportsService->buildDeceasedReportFhir($report, $this->getUser());
+                $response = $deceasedReportsService->createDeceasedReport($participantId, $fhirData);
+                $report = $report->loadFromFhirObservation($response);
+                $this->addFlash('success', 'Deceased Report created!');
```

Comment:
It looks like the form fields are still editable after the report as been created. We may need to redirect back to this route to make those fields disabled?
`return $this->redirectToRoute('deceased_report_new', ['participantId' => $participantId]);`

Reply:
Good point. It is also confusing because you get the success message and also the info alert that a report has already been created. Does the ticket or PRD say where the user should go after submitting the report? If not, I'm thinking it probably makes sense to redirect back to the participant summary page, since that is most likely where the user came from.

Reply:
Ah, that's the reason the "disabled" bit was in there earlier.

Redirecting to participant summary page is ideal, but we may have interesting logic to add with the discussion today about being able to initiate the process from different account types. For now, we can send it to `participant` route with the success flash message.

Reply:
Resolved in e4fe099f